File: packages/types/src/todo.ts
/**
 * TODO List Type Definition
 *
 * Define the core business types associated with TODO lists
 * for task tracking, progress management and agent collaboration
 */

import type { ID, Metadata } from "./common.js";

/**
 * TODO item status
 */
export type TodoStatus = "pending" | "in_progress" | "completed" | "cancelled";

/**
 * TODO Item Priority
 */
export type TodoPriority = "high" | "medium" | "low";

/**
 * TODO list item
 */
export interface TodoItem {
  /**
   * TODO ID
   * Should be unique within the session
   */
  id: ID;

  /**
   * TODO Content
   * Succinctly describe the task content
   */
  content: string;

  /**
   * TODO Status
   */
  status: TodoStatus;

  /**
   * Priority (optional)
   */
  priority?: TodoPriority;

  /**
   * Create timestamp (optional)
   */
  createdAt?: number;

  /**
   * Update timestamp (optional)
   */
  updatedAt?: number;

  /**
   * Metadata (optional)
   * Additional business information can be stored
   */
  metadata?: Metadata;
}

File: crates/wf-types/src/todo.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum TodoStatus {
    Pending,
    InProgress,
    Completed,
    Cancelled,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum TodoPriority {
    High,
    Medium,
    Low,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct TodoItem {
    pub id: super::Id,
    pub content: String,
    pub status: TodoStatus,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub priority: Option<TodoPriority>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub created_at: Option<i64>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub updated_at: Option<i64>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub metadata: Option<super::Metadata>,
}
